File: android/src/main/java/com/aspect/webviewbridge/protocol/BridgeModuleProtocol.kt
/**
 * WebView Bridge SDK - 模块协议
 *
 * 定义 Bridge 模块的基础接口
 */

package com.aspect.webviewbridge.protocol

// 注意：BridgeRequest 和 BridgeError 定义在 BridgeProtocol.kt 中

/**
 * Bridge 模块接口
 *
 * 所有能力模块必须实现此接口
 */
interface BridgeModule {
    /**
     * 模块名称
     */
    val moduleName: String
    
    /**
     * 支持的方法列表
     */
    val methods: List<String>
    
    /**
     * 处理请求
     *
     * @param method 方法名
     * @param request 请求对象
     * @param callback 回调函数
     */
    fun handleRequest(
        method: String,
        request: BridgeRequest,
        callback: (Result<Any?>) -> Unit
    )
}

/**
 * 模块生命周期接口
 *
 * 需要感知应用生命周期的模块可实现此接口
 */
interface BridgeModuleLifecycle {
    /**
     * 模块启用时调用
     */
    fun onEnable() {}
    
    /**
     * 模块禁用时调用
     */
    fun onDisable() {}
    
    /**
     * 应用进入前台
     */
    fun onResume() {}
    
    /**
     * 应用进入后台
     */
    fun onPause() {}
    
    /**
     * 销毁时调用
     */
    fun onDestroy() {}
}

/**
 * 模块上下文接口
 *
 * 提供模块所需的上下文信息
 */
interface BridgeModuleContext {
    /**
     * 发送事件到 Web 端
     */
    fun sendEvent(eventName: String, data: Any?)
    
    /**
     * 获取其他模块
     */
    fun <T : BridgeModule> getModule(moduleClass: Class<T>): T?
    
    /**
     * 获取 Activity （用于权限请求等操作）
     */
    fun getActivity(): android.app.Activity?
    
    /**
     * 获取 Bridge 实例（返回 Any 以避免循环依赖）
     * 实际返回类型应该是 WebViewBridge
     */
    fun getBridge(): Any?
}

/**
 * Result 扩展
 */
inline fun <T> Result<T>.fold(
    onSuccess: (T) -> Unit,
    onFailure: (Throwable) -> Unit
) {
    when {
        isSuccess -> onSuccess(getOrThrow())
        isFailure -> onFailure(exceptionOrNull()!!)
    }
}

/**
 * Bridge 回调接口
 * 提供简化的成功/错误回调
 */
interface BridgeCallback {
    fun success(data: Any?)
    fun error(error: BridgeError)
    
    companion object {
        /**
         * 从 Result 回调创建 BridgeCallback
         */
        fun from(callback: (Result<Any?>) -> Unit): BridgeCallback {
            return object : BridgeCallback {
                override fun success(data: Any?) {
                    callback(Result.success(data))
                }
                
                override fun error(error: BridgeError) {
                    callback(Result.failure(error))
                }
            }
        }
    }
}

/**
 * 将 (Result<Any?>) -> Unit 转换为 BridgeCallback
 */
fun ((Result<Any?>) -> Unit).toBridgeCallback(): BridgeCallback {
    return BridgeCallback.from(this)
}

File: webview-bridge-sdk/ios/Sources/Protocol/BridgeModuleProtocol.swift
/**
 * WebView Bridge SDK - 模块协议
 *
 * 定义 Bridge 模块的基础协议
 */

import Foundation

// MARK: - 模块协议

/// Bridge 模块协议
/// 所有能力模块都必须实现此协议
public protocol BridgeModule: AnyObject {
    /// 模块名称（必须唯一）
    var moduleName: String { get }
    
    /// 支持的方法列表
    var methods: [String] { get }
    
    /// 处理请求
    /// - Parameters:
    ///   - method: 方法名
    ///   - params: 请求参数
    ///   - callback: 完成回调
    func handleRequest(
        method: String,
        params: [String: AnyCodable],
        callback: @escaping (Result<Any?, BridgeError>) -> Void
    )
}

// MARK: - 默认实现

public extension BridgeModule {
    /// 检查是否支持某个方法
    func supportsMethod(_ method: String) -> Bool {
        return methods.contains(method)
    }
}

// MARK: - 模块上下文

/// 模块上下文，提供对 Bridge 核心功能的访问
public protocol BridgeModuleContext: AnyObject {
    /// 发送事件到 Web
    func sendEvent(_ event: BridgeEvent)
    
    /// 获取当前 WebView（如果需要）
    var webView: Any? { get }
}

// MARK: - 可配置模块

/// 可配置的模块协议
public protocol ConfigurableBridgeModule: BridgeModule {
    associatedtype Configuration
    
    /// 配置模块
    func configure(with configuration: Configuration)
}

// MARK: - 生命周期感知模块

/// 生命周期感知模块协议
public protocol LifecycleAwareBridgeModule: BridgeModule {
    /// Bridge 已准备就绪
    func onBridgeReady()
    
    /// Bridge 即将销毁
    func onBridgeWillDestroy()
    
    /// 应用进入前台
    func onAppWillEnterForeground()
    
    /// 应用进入后台
    func onAppDidEnterBackground()
}

// MARK: - 默认生命周期实现

public extension LifecycleAwareBridgeModule {
    func onBridgeReady() {}
    func onBridgeWillDestroy() {}
    func onAppWillEnterForeground() {}
    func onAppDidEnterBackground() {}
}
